src/core/net/srp_client.hpp

```diff
@@ -556,22 +556,36 @@ class Client : public InstanceLocator, private NonCopyable
     /**
      * This method starts the remove process of the host info and all services.
      *
-     * After retuning from this method, `Callback` will be called to report the status of remove request with
+     * After returning from this method, `Callback` will be called to report the status of remove request with
      * SRP server.
      *
      * If the host info is to be permanently removed from server, @p aRemoveKeyLease should be set to `true` which
      * removes the key lease associated with host on server. Otherwise, the key lease record is kept as before, which
      * ensures that the server holds the host name in reserve for when the client once again able to provide and
      * register its service(s).
      *
-     * @param[in] aRemoveKeyLease  A boolean indicating whether or not the host key lease should also be removed.
+     * The @p aSendUnregToServer determines the behavior when the host info is not yet registered with the server. If
+     * @p aSendUnregToServer is set to `false` (which is the default/expected value) then the SRP client will
+     * immediately remove the host info and services without sending an update message to server (no need to update the
+     * server if nothing is yet registered with it). If @p aSendUnregToServer is set to `true` then the SRP client will
+     * send an update message to the server. Note that if the host info is registered then the value of
+     * @p aSendUnregToServer does not matter and the SRP client will always send an update message to server requesting
+     * removal of all info.
+     *
+     * One situation where @p aSendUnregToServer can be useful is on a device reset/reboot, caller may want to remove
+     * any previously registered services with the server. In this case, caller can `SetHostName()` and then request
+     * `RemoveHostAndServices()` with `aSendUnregToServer` as `true`.
+     *
+     * @param[in] aRemoveKeyLease     A boolean indicating whether or not the host key lease should also be removed.
+     * @param[in] aSendUnregToServer   A boolean indicating whether to send update to server when host info is not
+     *                                registered.
      *
      * @retval kErrorNone      The removal of host and services started successfully. The `Callback` will be called
      *                         to report the status.
      * @retval kErrorAlready   The host is already removed.
      *
      */
-    Error RemoveHostAndServices(bool aShouldRemoveKeyLease);
+    Error RemoveHostAndServices(bool aShouldRemoveKeyLease, bool aSendUnregToServer = false);
```

Comment:
```suggestion
    Error RemoveHostAndServices(bool aShouldRemoveKeyLease, bool aForceUnregToServer = false);
```

nit: according to the document, seems this is an argument to **force** unreg to server.

Reply:
I actually first used the term "force" in the name but changed it to the current name later.

The main reason is `aForceUnregToServer` makes it sound as if we are "forcing the removal of the entry on the SRP server" (which is not really the case). We cannot from client side force this on server. All we are doing is ensuring that from client side we are sending the request message to server (sever can still reject the removal request, e.g. if key is not correct).

Reply:
`force` can also mean force the SRP client to send the unregister (i.e. send unregister)?
Since it's an SRP client API, users should already know that it can't force SRP servers to do anything?

Reply:
It is easy to associate "Force" with the API name itself which is "RemoveHostAndService" making it sound like it is "forcing remove" (on server or client) which as I said is not what is happening.

Reply:
SGTM